scripts/repoHandlers.js

```diff
@@ -72,40 +72,57 @@ function updateTrackedReposInBitbucketOrg(trackedRepos, websiteUrl) {
  * @param {boolean} isOrg - Whether github organization or user.
  */
 function updateTrackedReposInGitHub(trackedRepos, websiteUrl, ownerType) {
-	const allRepo = ownerType == 'org' ? document.getElementById('org-repositories') : document.getElementById('user-repositories-list');
-	const repoUrl = Array.from(allRepo.querySelectorAll('a[itemprop="name codeRepository"]'));
+    let repoList;
+    let newUI = false;
 
+    // Check if selectors for new UI exist
+    if (document.querySelector('[data-testid="list-view-items"]')) {
+        repoList = ownerType === 'org' ? document.querySelectorAll('[data-testid="list-view-items"] > li') : document.querySelectorAll('[data-filterable-for="your-repos-filter"] > li');
+		newUI = true;
+    } else {
+        const allRepo = ownerType === 'org' ? document.getElementById('org-repositories') : document.getElementById('user-repositories-list');
+        repoList = Array.from(allRepo.querySelectorAll('a[itemprop="name codeRepository"]'));
+    }
+    repoList.forEach(repoItem => {
+		let repoUrl;
+		let repoLink
+		if (newUI){
+			repoLink = ownerType == 'org' ? repoItem.querySelector('[data-testid="listitem-title-link"]') : repoItem.querySelector('a[itemprop="name codeRepository"]');
+			if (!repoLink) return;
```

Comment:
The code of the new UI and the old UI are highly coupled here. It is hard to read this and it will be hard to remove the code for the old UI when it is no more in beta (or the new UI if GitHub decides to rollback).